lib/Dialect/TritonGPU/Transforms/Pipeline.cpp

```diff
@@ -1612,15 +1614,92 @@ void PipelinePass::updateConsumerReleaseInfo(Operation *oldOp, Operation *newOp,
   }
 }
 
+int PipelinePass::getDependencyIndex(Value v, scf::ForOp forOp) {
+  auto op = v.getDefiningOp();
+  // root and not BlockArgument
+
+  auto iterArgs = forOp.getInitArgs();
```

Comment:
I have a little misunderstanding. Do you have a concrete example to explain why we check `forOp.getInitArgs()` here but not `forOp.getRegionIterArgs()`?